### PR TITLE
Added player abilities

### PR DESCRIPTION
**Pull request by Castelias**
Added new Halo ultimate and homing missile

### CODE REVIEW

File: Assets/Abilities.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        _playerInput.Disable();
+    }
+    
+
+
+
+    private IEnumerator UltimateMissile()
+    {
+        for (int i = 0; i <= 9; i++)
```
Comment (PeterPwlk) [inline Assets/Abilities.cs:41]: Make number of missiles configurable

File: Assets/changeSize.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DefaultNamespace
+{
+    public class changeSize : MonoBehaviour
```
Comment (PeterPwlk) [inline Assets/changeSize.cs:7]: Class name should start with Capital letter

File: Assets/changeSize.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        Vector2 temp;
```
Comment (PeterPwlk) [inline Assets/changeSize.cs:9]: If field is used only inside a class make it private

File: Assets/changeSize.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        public float changingSpeed = 10f;
+        public int haloDamage = 200;
```
Comment (PeterPwlk) [inline Assets/changeSize.cs:11]: If fields are changed in inspector make them private and add [SerializeField] attribute

File: Assets/changeSize.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+        private IEnumerator Start()
+        {
+            yield return new WaitForSeconds(2);
```
Comment (PeterPwlk) [inline Assets/changeSize.cs:15]: Make interval configurable

File: Assets/changeSize.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            temp.x += changingSpeed * Time.deltaTime;
+            temp.y += changingSpeed * Time.deltaTime;
+
+            transform.localScale = temp;
+
+        }
+        private void OnTriggerEnter2D(Collider2D hitInfo)
+        {
+
+            Enemy enemy = hitInfo.gameObject.GetComponent<Enemy>();
```
Comment (PeterPwlk) [inline Assets/changeSize.cs:32]: What is enemy variable used for ?

File: Assets/missile.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+
+
+
+namespace DefaultNamespace
+{
+    [RequireComponent(typeof(Rigidbody2D))]
+    public class missile : MonoBehaviour
```
Comment (PeterPwlk) [inline Assets/missile.cs:11]: Class name should start with Capital letter

File: Assets/missile.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        public Transform target;
+        public float speed = 5f;
+        public float rotateSpeed = 200f;
+        private Rigidbody2D rb;
+        public int missileDamage = 100;
```
Comment (PeterPwlk) [inline Assets/missile.cs:17]: As before make fields private and add [SerializeField] attribute

File: Assets/missile.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace DefaultNamespace
+{
+    [RequireComponent(typeof(Rigidbody2D))]
+    public class missile : MonoBehaviour
+    {
+        public Transform target;
+        public float speed = 5f;
+        public float rotateSpeed = 200f;
+        private Rigidbody2D rb;
```
Comment (PeterPwlk) [inline Assets/missile.cs:16]: Private fields without [SerializeField] should start with _

File: Assets/missile.cs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            rb.velocity = transform.up * speed;
+
+        }
+        private void OnTriggerEnter2D(Collider2D hitInfo)
+        {
+            var health = hitInfo.gameObject.GetComponent<HealthManager>();
+            health?.DealDamage(missileDamage);
+
+            Enemy enemy = hitInfo.GetComponent<Enemy>();
+            if (enemy != null)
```
Comment (PeterPwlk) [inline Assets/missile.cs:44]: Shouldn't the missile be destoyed if it collides with obstacles ?

File: Assets/Abilities.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class Abilities : MonoBehaviour
```
Comment (PeterPwlk) [inline Assets/Abilities.cs:7]: Maybe PlayerAbilitiesController would be more descriptive name ?

File: Assets/Abilities.cs [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+{
+    public Transform ShootingPoint_Mid;
+    public GameObject missilePrefab;
+    private PlayerInput _playerInput;
+    public GameObject haloPrefab;
```
Comment (PeterPwlk) [inline Assets/Abilities.cs:12]: [SerializeField] and private

File: Assets/Scripts/PlayerAbilitiesController.cs [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        _playerInput.Disable();
+    }
+    
+
+
+
+    private IEnumerator UltimateMissile()
+    {
+        for (MissileAmount = 1; MissileAmount <= 9; MissileAmount++)
```
Comment (PeterPwlk) [inline Assets/Scripts/PlayerAbilitiesController.cs:44]: That's hell of a config XD

Comment (Castelias) [reply Assets/Scripts/PlayerAbilitiesController.cs:44]: xD

**Pull request merged**